performance_metrics/tutorial.md

```diff
@@ -0,0 +1,187 @@
+defined in the world. In this case we expect a `sim_sensor_update_rate` equals to 500 in the high resolution camera
+and 30 in the low resolution. Then the `fps` value means that Gazebo is able to generate 337 frames in a *real second*.
+
+The real time factor is likely less than 1.0. The exact number depends on your computing power. This shows that
+the sensor's update rate is strictly followed, and physics has slowed down in order to accommodate for the high update rate.
+
+```bash
+real_time_factor: 0.5083145
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor"
+  real_sensor_update_rate: 254.39207924822054
+  sim_sensor_update_rate: 500
+  fps: 279.38592529296875
+}
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor_regular"
+  real_sensor_update_rate: 12.626295945654704
+  sim_sensor_update_rate: 30.3030303030303
+  fps: 16.778518676757812
+}
+```
+
+## Contact sensor
+
+For the physics-based sensors we are going to use the [contact_strict_rate.world](https://github.com/osrf/gazebo/blob/gazebo9/test/worlds/contact_strict_rate.world) world. It contains a contact sensor (1000 Hz).
+
+In this case both world look pretty similar, even with this high update rate. Because the update loop
+is small in this case. This may vary with a more complex object
+
+### Run the world without lockstep
+
+```bash
+real_time_factor: 0.988433204
+sensor {
+  sensor_name: "world_1::model_1::link_1::sensor_contact"
+  real_sensor_update_rate: 964.34805249910789
+  sim_sensor_update_rate: 1000
+}
+```
+
+### Run the world with lockstep
+
+```
+real_time_factor: 1.003396497
+sensor {
+  sensor_name: "world_1::model_1::link_1::sensor_contact"
+  real_sensor_update_rate: 1006.3885545442469
+  sim_sensor_update_rate: 1000
+}
+```
+
+## Real time update rate
+
+We can modify the Real time update rate in the Physics Engine. As you may see the `sim_sensor_update_rate` in both cameras
+are the expected value 500 and 30. In this case the Real time update rate is equal to **10** which make the simulation slower. The fps
+are lower, it's able to generate almost 6 frames for the high speed camera and 2 for the regular camera each *real second*.
+
+### Run the world with lockstep
+
+```bash
+real_time_factor: 0.969638674
+sensor {
+  sensor_name: "world_1::model_1::link_1::sensor_contact"
+  real_sensor_update_rate: 972.45520628206054
+  sim_sensor_update_rate: 1000
+}
+
+real_time_factor: 1.008738703
+sensor {
+  sensor_name: "world_1::model_1::link_1::sensor_contact"
+  real_sensor_update_rate: 1011.0170528246299
+  sim_sensor_update_rate: 1000
+}
+```
+
+### Run the world without lockstep
+
+Again in this case, where we run the world without `lockstep`, there is no guarantee that the camera sensor update happens
+in the same iteration as the physics simulation, and the `sim_sensor_update_rate` value is highly impacted, it should be
+close to 500 but in this case it's close to 0.
```

Comment:
the description here describes update rates for camera sensors but the values below are from contact sensor